### PR TITLE
handleCreatePost

### PR DESCRIPTION
**Pull request by key-er**
we didn't have "title" column so added that and changed schema.

image

image

### CODE REVIEW

File: pages/requiredfields.js [hunk 1]
```diff
@@ -95,15 +97,52 @@ function RequiredFields(props) {
...
-    // Handle saving changes
+  const handleSaveChanges = (e) => {
+      e.preventDefault();
+      const response = fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          emailAddr: document.getElementsByClassName("jsx-db670790b21b4b04 top display text-light").valueOf()[0].innerText.split(': ')[1].split(' (')[0],
```
Comment (ajbieber) [inline pages/requiredfields.js]: You can get this value from the props using `props.user.email`

File: pages/requiredfields.js [hunk 2]
```diff
@@ -95,15 +97,52 @@ function RequiredFields(props) {
...
     setShowModal(true);
     setEditMode(true);
+    
+    // const apiUrl = 'http://localhost:3000/api/landingPage'; 
+    if (post == "") {
+      console.log("\n empty content is not allowed");
+      return
+    }
+    const addConfirmation = fetch(url, {
+      method: "POST",
```
Comment (ajbieber) [inline pages/requiredfields.js]: Would this be a `PATCH` since we are doing an update

File: pages/requiredfields.js [hunk 3]
```diff
@@ -95,15 +97,52 @@ function RequiredFields(props) {
...
+      body: JSON.stringify({
+        id: post.id,
+        content: post.content,
+      }),
+    });
+    if (addConfirmation.status !== 200) {
+      console.log("something went wrong");
+    } else {
+      console.log("post updated successfully !!!");
+    }
```
Comment (ajbieber) [inline pages/requiredfields.js]: I don't think you want this logic here. This function is called when the edit button is clicked. Ideally the flow should be the following:

1. When the user clicks the edit button, a modal appears
2. The modal is prefilled with the posts data
3. When the "Update" button is clicked, the request is sent to update the content. 

I think you are going to want to put this content in the `handleSaveChanges` method. If `editMode === true`, you take the content and PATCH it. If `editMode === false` your POST the data.

File: pages/requiredfields.js [hunk 4]
```diff
@@ -1436,8 +1475,11 @@ function RequiredFields(props) {
                         as="textarea"
                         rows={3}
                         placeholder="Enter content"
-                        value={content}
-                        onChange={(e) => setContent(e.target.value)}
+                        // value={userContent}
```
Comment (ajbieber) [inline pages/requiredfields.js]: You do need this value uncommented. This is necessary displaying the value. 

Every time the user types on their keyboard, the `onChange` method is called. Lets say the following situation happens. Currently, the textarea is empty, and I want to type "Hello"

With the current implementation, when I enter "H", we will call `setUserContent("H")`. But, because the value is never set, it will remain a blank string so the user won't see the value typed in the text box.

File: pages/requiredfields.js [hunk 5]
```diff
@@ -1448,7 +1490,8 @@ function RequiredFields(props) {
                     Close
                   </Button>
                   {editMode ? (
-                    <Button variant="primary" onClick={handleSaveChanges}>
```
Comment (ajbieber) [inline pages/requiredfields.js:1451]: Why did you remove the onClick method? Without this, we wont call the method when the user clicks the button.

File: pages/requiredfields.js [hunk 6]
```diff
@@ -67,7 +67,8 @@ function RequiredFields(props) {
   const [editMode, setEditMode] = useState(false);
   const [selectedPost, setSelectedPost] = useState(null);
   const [confirmDelete, setConfirmDelete] = useState(false);
-
+  const [userContent, setUserContent] = useState('');
+  const [userPosts, setUserPosts] = useState('');
```
Comment (ajbieber) [inline pages/requiredfields.js]: Not sure what this is used for.

**Review by ajbieber — CHANGES_REQUESTED** [review]

File: pages/requiredfields.js (hunk 2 above)
Comment (key-er) [reply pages/requiredfields.js]: no it is adding a new post technically the button should say create post not save changes

File: pages/requiredfields.js (hunk 5 above)
Comment (key-er) [reply pages/requiredfields.js:1451]: it works I had it in two places by mistake

File: pages/requiredfields.js (hunk 3 above)
Comment (key-er) [reply pages/requiredfields.js]: These names were already there and I know it is very misleading. It should have named as handleCreatePost
and the button should have said "Create" instead of "Save Changes" -  All changed now

**Review by ajbieber — APPROVED** [review]
